File: src/leetCode/Ex832.java
package leetCode;

public class Ex832 {
    public int[][] flipAndInvertImage(int[][] A) {
        for(int i = 0;i<A.length;i++){
            int len = A[i].length;
            for(int j = 0;j<len/2;j++){
                //swap A[i][j] and A[i][l-j]
                //and flip bit
                int t = A[i][j];
                A[i][j] = A[i][len-j-1] ^ 1;
                A[i][len-j-1] = t ^ 1;
            }
            // odd case, the middle one shoud be fliped
            if(len%2!=0){
                A[i][len/2] ^= 1;
            }
        }
        return A;
    }

    public static void main(String[] args) {
        var o = new Ex832();
        o.flipAndInvertImage(new int[][]{{1,1,0},{0,0,1},{0,0,0}});
    }
}

File: src/leetCode/ex832.kt
package leetCode

fun flipAndInvertImage(A: Array<IntArray>): Array<IntArray> {
    for(a in A){
        val len = a.size/2
        for (i in 0 until len){
            swap(a,i,a.size-i-1)
        }
        for (i in 0 until a.size){
            if (a[i]==0){
                a[i]=1
            }else{
                a[i]=0
            }
        }
    }
    return A
}

fun swap(array:IntArray,l:Int,r:Int){
    val temp = array[r]
    array[r]=array[l]
    array[l]=temp
}

fun main(args: Array<String>) {
    val a = arrayOf(intArrayOf(1,1,0), intArrayOf(1,0,1), intArrayOf(0,0,0))
    flipAndInvertImage(a)
    a.forEach { it.forEach(::print);println() }
}